File: JAVA DSA/InsertionSort.java
public class InsertionSort {

    public static void main(String[] args) {
        int[] arr = { 5, 4, 3, 2, 1 };

        for (int i = 0; i <= arr.length - 1; i++) {
            int j = i;
            while (j > 0 && arr[j - 1] > arr[j]) {
                int temp = arr[j-1];
                arr[j-1] = arr[j];
                arr[j] = temp;
                j--;
            }
        }
        for (int ele : arr)
            System.out.print(ele + " ");

    }
}

File: C PROGRAMS/DAA 3rd Sem/InsertionSort.c
#include<stdio.h>

void insertionSort(int arr[],int n){

    for(int i=1;i<n;i++){
        int key = arr[i];
        int j = i-1;
        while(j>=0 && arr[j]>key){
            arr[j+1] = arr[j];
            j--;
        }
        arr[j+1] = key;
    }

}

int main(){

    int arr[]={10,9,8,7,6,5,4,3,2,1};
    int n = 10;

    insertionSort(arr,n);

    for(int i=0;i<n;i++){
        printf("%d ",arr[i]);
    }

    return 0;
}